### DISCUSSION TITLE
Change file attributes in PNP results

### DISCUSSION THREAD

**Discussion by Obsouleet** [microsoft-search/pnp-modern-search#4115] [Q&A] [answered]
Hello everybody,

I'm currently building a document library for my team. To my great joy I found **PNP modern search** which made my approach of "putting all files on a huge pile and filter them by meta data" feasible. Everybody liked it: 

![Screenshot1](https://github.com/user-attachments/assets/6cdfa6a3-f418-4579-8295-230bb496dbc5)

Next I wanted my colleagues to be able to edit the file properties (name and list columns) directly from this view. Unfortunately I could not find anyone implementing such a task.

I can't believe that there is no way to call the internal sharepoint routines for "rename" or "editProps".  

Will I need to have two front ends for my library - one to filter and one to manage the files?

**Comment by kasperbolarsen** [comment, marked as answer]
We are actively blocking custom code in the layouts for safety reasons so that would not be possible

**Reply by wobba** [reply]
Any interaction with API calls has to be added as custom web components. https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/
